dns/miekgdns2/resolver_test.go

```diff
@@ -0,0 +1,323 @@
+package miekgdns2
+
+import (
+	"context"
+	"errors"
+	"net"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/miekg/dns"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+)
+
+var resolvConfContents = `
+nameserver 127.0.0.1
+nameserver 127.0.0.2
+option attempts:2
+option timeout:1
+`
+
+func TestResolver_LookupSRV(t *testing.T) {
+	t.Run("bad resolv.conf", func(t *testing.T) {
```

Comment:
[nit] This might be a clearer name for the test:

```suggestion
	t.Run("missing resolv.conf", func(t *testing.T) {
```